tiledb/sm/query/reader.cc

```diff
@@ -3161,6 +3168,43 @@ Status Reader::copy_attribute_values(
   STATS_END_TIMER(stats::Stats::TimerType::READ_COPY_ATTR_VALUES);
 }
 
+Status Reader::add_extra_offset() {
+  for (const auto& it : buffers_) {
+    const auto& name = it.first;
+    // TODO: keep that check?
+    if (read_state_.overflowed_)
+      break;
+
+    if (!array_schema_->is_attr(name) || !array_schema_->var_size(name))
+      continue;
+
+    if (*it.second.buffer_size_ >= it.second.original_buffer_size_) {
```

Comment:
This only checks that we have at least one byte, but we need sizeof(uint64_t) bytes. How about:
```
if (*it.second.buffer_size_ + sizeof(uint64_t) > it.second.original_buffer_size_) {
```